Data for Quests

jmgoldb opened the pull request:
I retroactively split up JSON changes and code changes into separate branches so I could have two separate pull requests.

Included with these JSON files is the new prebuilt store, which is incompatible with the model/code in the app, so this pull request will fail on travis.  The next pull request I have after this one should align things back together and work again. 

With that said, can I get a quick review, please? @vokal/ios-developers

jmgoldb committed:
Data for Quests

jmgoldb committed:
Quest Lists are functional (Quest Details up next)

brockboland commented:
Why not combine all the related changes in a single PR?

jmgoldb commented:
I didn't want to bloat a PR with actual code changes with 5k lines of find-replace json changes.  But I can merge in the other branch if that'd be preferable.

jmgoldb committed:
Merge branch 'questHandling' into questData

brockboland commented on MH4U Dex/CoreDataController.m:
```diff
@@ -68,8 +69,8 @@ - (void)tryLoadSequence
-        [self attemptSaveContext:backgroundContext];
-        NSLog(@"Core Data Context Saved.");
+//        [self attemptSaveContext:backgroundContext];
+//        NSLog(@"Core Data Context Saved.");
```
Commented-out code can be removed

jmgoldb replied:
I need to leave that in because I still use it whenever I need to change the prebuilt store.

brockboland commented on MH4U Dex/CoreDataController.m:
```diff
@@ -53,7 +54,7 @@ - (void)tryLoadSequence
         NSLog(@"Resetting Core Data");
         [self resetCoreData];
-        NSManagedObjectContext *backgroundContext = [self backgroundManagedObjectContext];
+//        NSManagedObjectContext *backgroundContext = [self backgroundManagedObjectContext];
```
Commented out code can be removed

jmgoldb replied:
Same as above

brockboland replied:
Why don't you move all that into a method, and only add a call to the method as needed?

jmgoldb replied:
That sounds like a good idea!

brockboland commented on MH4U Dex/CoreDataController.m:
```diff
@@ -167,41 +168,51 @@ - (void)loadQuestDataWithContext:(NSManagedObjectContext *)context
+    for (NSURL *url in urlArray)  {
+        NSString *fileName = [[url URLByDeletingPathExtension] lastPathComponent];
+        if ([fileName hasPrefix:QuestDropFilePrefix]) {
+            NSLog(@"Loading in quest drop file: %@", fileName);
```
Do you want to keep this debug logging?

jmgoldb replied:
Yes, for now it helps me figure out if I've stalled loading a single file or not.

brockboland commented on MH4U Dex/CoreDataController.m:
```diff
@@ -167,41 +168,51 @@ - (void)loadQuestDataWithContext:(NSManagedObjectContext *)context
+                NSPredicate *questPredicate = [NSPredicate predicateWithFormat:@"%K == %@", QuestAttributes.id, questDropDict[MHDQuestDropQuestIDKey]];
+                Quest *quest = (Quest *)[self uniqueEntityWithEntityName:[Quest entityName]
+                                                  withPredicate:questPredicate
+                                                    withContext:context];
```
Colons aren't lined up here

brockboland commented on MH4U Dex/CoreDataController.m:
```diff
@@ -472,29 +484,70 @@ - (Quest *)questFromDictionary:(NSDictionary *)questDict withContext:(NSManagedO
-        if (quest.secondMonster) {
+        monster = (Monster *)[self uniqueEntityWithEntityName:[Monster entityName]
+                                                withPredicate:monsterPredicate
+                                                  withContext:context];
```
I wouldn't re-use the `monster` variable here when loading a different object, just to make sure you don't accidentally use the wrong one later on.

brockboland replied:
Same thing about the following couple of cases

brockboland commented on MH4U Dex/CoreDataController.m:
```diff
@@ -472,29 +484,70 @@ - (Quest *)questFromDictionary:(NSDictionary *)questDict withContext:(NSManagedO
+//                }
+//            }
+//        }
+//    }
```
When you do put this back in: same thing about re-using a variable name.

brockboland commented on MH4U Dex/CoreDataController.m:
```diff
@@ -571,6 +624,7 @@ - (NSPersistentStoreCoordinator *)persistentStoreCoordinator
     }
     
     NSURL *storeURL = [self.applicationDocumentsDirectory URLByAppendingPathComponent:@"MH4U_DEX.sqlite"];
+//    NSLog(@"%@", storeURL);
```
Commented code can be removed

jmgoldb replied:
Again, I'm going to leave that in for now, because I'll need to uncomment it in the near future.

jmgoldb committed:
Addressed code review

jmgoldb commented:
Split the monster variable into four different variables, and fixed the colon indentation

brockboland commented on MH4U Dex/CoreDataController.m:
```diff
@@ -472,29 +484,70 @@ - (Quest *)questFromDictionary:(NSDictionary *)questDict withContext:(NSManagedO
-            if (quest.thirdMonster) {
+        if (secondMonster) {
+            [quest.monsterSet addObject:secondMonster];
+            monsterPredicate = [NSPredicate predicateWithFormat:@"%K == %@", MonsterAttributes.name, questDict[MHDQuestThirdMonsterKey]];
```
You addressed the `monster` variable, but you're still re-using the `monsterPredicate` throughout this method. I'd recommend changing that too.

brockboland commented:
Regarding your last commit: I know it's been pretty common for people to make commits with messages like "handle code review" and such, but that won't help you identify what the commit actually did when you're skimming the commit list later on. Try to use commit messages that actually explain what was changed or why.

jmgoldb committed:
Moved commented out lines to handle rebuilding a prebuilt store to a separate function, and split monsterPredicate into four pointers when assigning monsters to a quest

Also added a TODO to remind myself to do the same once I reimplement quest prerequisites.

jmgoldb commented:
You're right, I'll try and catch myself before doing that

jmgoldb commented:
How does it look now?

brockboland commented:
Much better! Good to pull!

jmgoldb commented:
Thanks!   @seanwolter @chillpop @designatednerd May I get a merge, please?

jmgoldb commented:
Thanks
